ui/app/adapters/pki/certificate/base.js

```diff
@@ -4,10 +4,46 @@ import ApplicationAdapter from '../../application';
 export default class PkiCertificateBaseAdapter extends ApplicationAdapter {
   namespace = 'v1';
 
-  deleteRecord(store, type, snapshot) {
+  getURL(backend, id) {
+    const uri = `${this.buildURL()}/${encodePath(backend)}`;
+    return id ? `${uri}/cert/${id}` : `${uri}/certs`;
+  }
+
+  fetchByQuery(query) {
+    const { backend, id } = query;
+    const data = !id ? { list: true } : {};
+    return this.ajax(this.getURL(backend, id), 'GET', { data }).then((resp) => {
+      resp.data.backend = backend;
+      if (id) {
+        resp.data.id = id;
+        resp.data.serial_number = id;
+      }
+      return resp;
+    });
+  }
+
+  query(store, type, query) {
+    return this.fetchByQuery(query);
+  }
+
+  queryRecord(store, type, query) {
+    return this.fetchByQuery(query);
+  }
+
+  // the only way to update a record is by revoking it which will set the revocationTime property
+  updateRecord(store, type, snapshot) {
```

Comment:
This was originally destroying the record but it's actually an update since the `recovation_time` property is set and the certificate is still returned from the server.